src/pipx/commands/install.py

```diff
@@ -33,73 +33,86 @@ def install(
 
     python = python or DEFAULT_PYTHON
 
-    if package_name is None:
-        package_name = package_name_from_spec(package_spec, python, pip_args=pip_args, verbose=verbose)
-    if venv_dir is None:
-        venv_container = VenvContainer(constants.PIPX_LOCAL_VENVS)
-        venv_dir = venv_container.get_venv_dir(f"{package_name}{suffix}")
+    if package_names is None:
+        package_names = [
+            package_name_from_spec(package_spec, python, pip_args=pip_args, verbose=verbose)
+            for package_spec in package_specs
+        ]
+    elif len(package_names) != len(package_specs):
```

Comment:
This seems to hold exactly the same logic as the condition above. Could you merge the two?